src/DynamoCore/Search/SearchDictionary.cs

```diff
@@ -379,12 +380,13 @@ internal IEnumerable<V> Search(string query, int minResultsForTolerantSearch = 0
             query = query.ToLower();
 
             var subPatterns = SplitOnWhiteSpace(query);
-
-            // Add full (unsplit by whitespace) query to subpatterns
-            var subPatternsList = subPatterns.ToList();
-            subPatternsList.Insert(0, query);
-            subPatterns = (subPatternsList).ToArray();
-
+            if (subPatterns.Length > 1)// More than one word
+            {
+                // Add full (unsplit by whitespace) query to subpatterns
+                var subPatternsList = subPatterns.ToList();
+                subPatternsList.Insert(0, query);
+                subPatterns = (subPatternsList).ToArray();
```

Comment:
Could you explain with an example? If the query is `all elements of category`, the subpatterns list will be `["all", "elements", "of", "category"]`, then the subpatterns list will now be `["all elements of category", "all", "elements", "of", "category"]`?

Reply:
that is correct.
I think the logic behind it was to look for the full text first (a match with the full text should have the highest priority).